### ISSUE TITLE
Using busy-waiting instead of waiting for wakeup

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/ucx-py#367]
Based on discussion at the UCF Hackathon, OpenUCX developers relayed that waiting for wakeup is slow and that busy-waiting is more performant. Not sure to what extent we have explored this, but it may be worth looking at using busy-waiting and seeing what sort of performance improvement this would give us.

**Comment by madsbk** [comment]
Setting `UCXPY_NON_BLOCKING_MODE=1` should make ucxpy use busy-waiting

**Comment by jakirkham** [comment]
IIRC we are still polling in Python with the GIL held. Is that correct? I'm wondering if we lower it and release the GIL if that will help.

**Comment by madsbk** [comment]
Yes, that it a good point. Also, we might try to have a dedicated progress thread.

**Comment by madsbk** [comment]
Yes, that it a good point. Also, we might try to have a dedicated progress thread.

**Comment by jakirkham** [comment]
That's a good point. I wonder if we should also consider a thread for handling callbacks as well.

**Comment by pentschev** [comment]
UCXX supports all modes, blocking, polling, wait and also blocking/polling in a dedicated progress thread. Closing.
